### ISSUE TITLE
When using popup method, hide the line under the pum entirely

### ISSUE THREAD

**Issue by Aster89** [andymass/vim-matchup#179]
**Is your feature request related to a problem? Please describe.**
When `let g:matchup_matchparen_offscreen = {'method': 'popup', 'highlight': 'OffscreenPopup'}` is set, the fact that the popup doesn't extend for the whole width of the window can be confusing.

**Describe the solution you'd like**
Maybe hiding the rest of the line on which the popup appears could be a good solution. And an option could let the user decide to opt in.

**Describe alternatives you've considered**
Well, I can live with it. It's just a bit annoying

**Additional context**
If the scenario is

  - `let g:matchup_matchparen_offscreen = {'method': 'popup', 'highlight': 'OffscreenPopup'}`
  - Top line on screen is `namespace abc { namespace def {` (I'm referring to a C++ file)
  - Bottom line on screen is the closing `}` of a `namespace xyz {` that comes before the top line on screen
  - The cursor is that closing `}` on the bottom line on screen.

then the line on top results someting like `namspace xyz { namespace def {`, which can be a bit confusing.

**Comment by andymass** [comment]
There was already an option `fullwidth` in neovim but it was not allowed for vim.

    let g:matchup_matchparen_offscreen = {'method': 'popup', 'fullwidth': 1, ...}

It makes sense to add, so done.  Please let me know how it works.

**Comment by Aster89** [comment]
Thanks, that seems to work.

I have a side question that's independent form this change. I've noticed that the syntax coloring is off when `method` is `popup` and on when `method` is `status`. Is this expected and unavoidable?

**Comment by andymass** [comment]
Is your question why the syntax highlighting is different in the status line than the popup?  Yeah, it's just not implemented yet, but can possibly be by re-using the same code as the status method.

**Comment by Aster89** [comment]
Yeah, the difference is what surprises me a bit.

**Comment by andymass** [comment]
Highlighting in popup added as of 0a93cee48d45f8d7b61aad288434ed48527727e5

**Comment by andymass** [comment]
Note this is currently behind an option.   Please use

let matchup_matchparen_offscreen={'method': 'popup', 'syntax_hl': 1}

**Comment by Aster89** [comment]
Is it normal that I see the syntax coloring without `'syntax_hl': 1` and even with `'syntax_hl': 0`?

**Comment by andymass** [comment]
I think also related to update.  Now, with syntax_hl = 0, only the match
word should be highlighted.  With syntax_hl = 1, the whole line will be
syntax highlighted.  The option was added as there may be performance
implications.
